vllm/outputs.py

```diff
@@ -223,7 +223,11 @@ def from_seq_group(
             if delta:
                 # Slice logprobs delta if applicable
                 if output_logprobs:
-                    output_logprobs = output_logprobs[-num_output_tokens:]
+                    # Can happen when n > 0 and request finishes before the othere
+                    if num_output_tokens == 0:
+                        output_logprobs = None
+                    else:
+                        output_logprobs = output_logprobs[-num_output_tokens:]
```

Comment:
nit: invert so that "normal" case is first?
```suggestion
                    if num_output_tokens:
                        output_logprobs = output_logprobs[-num_output_tokens:]
                    else:
                        output_logprobs = None
```